src/gui/editors/PianoRoll.cpp

```diff
@@ -4145,36 +4145,34 @@ void PianoRoll::pasteGhostNotes()
 	if(!hasValidPattern()) { return; }
 
 	QString value = Clipboard::getString(Clipboard::MimeType::Default);
+	QDomNodeList list;
 
 	if(!value.isEmpty())
 	{
 		DataFile dataFile(value.toUtf8());
+		list = dataFile.elementsByTagName(Note::classNodeName());
+	}
 
-		QDomNodeList list = dataFile.elementsByTagName(Note::classNodeName());
-
-		if(!list.isEmpty())
-		{
-			m_ghostNotes.clear();
+	if(list.isEmpty())
+	{
+		TextFloat::displayMessage(tr("Paste as ghost failed"),
```

Comment:
Should this return after this or continue clearing the ghost notes and all that?

Reply:
Oh sorry I forgot to add return there